db/migrate/20150514114044_migrate_ptables_to_templates.rb

```diff
@@ -0,0 +1,84 @@
+class MigratePtablesToTemplates < ActiveRecord::Migration
+  class FakeOldPtable < ActiveRecord::Base
+    self.table_name = 'ptables'
+
+    has_and_belongs_to_many :operatingsystems, :join_table => 'operatingsystems_ptables', :foreign_key => 'ptable_id'
+    has_many :hostgroups, :foreign_key => 'ptable_id'
+    has_many_hosts :foreign_key => 'ptable_id'
+  end
+
+  class FakeNewPtable < ActiveRecord::Base
+    self.table_name = 'templates'
+
+    has_and_belongs_to_many :operatingsystems, :join_table => 'operatingsystems_ptables', :foreign_key => 'ptable_id'
+    has_many :hostgroups, :foreign_key => 'ptable_id'
+    has_many_hosts :foreign_key => 'ptable_id'
+  end
+
+  def up
+    remove_foreign_key "operatingsystems_ptables", :name => "operatingsystems_ptables_ptable_id_fk"
```

Comment:
These all need guarding with checks to ensure the fkeys exist, they don't in all databases.